### ISSUE TITLE
Improve VS Code when working with network drives

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#13665]
Some issues impact users that try to run VS Code against slow network drives. Specifically:
- [ ] some folders fail to open or display (#26931, #20207, #21397, #21520, #25885)
- [ ] we remove folders from the recent folders list when they do not seem to exist even though they are ([13699](https://github.com/Microsoft/vscode/issues/13699))
- [x] when opening an already opened file, we load its metadata to find out if it changed (#5353)
- [x] when opening an already opened file, we open it to detect the encoding (https://github.com/Microsoft/vscode/issues/21604)
- [x] we close an editor when we detect a file delete event (this can happen from faulty file events) (#11642)
- [x] we prevent saving when we think the contents on disk are newer (which seems to happen sporadically on network drives) (#18037)

**Comment by dogmatic69** [comment]
> we remove folders from the recent folders list when they do not seem to exist even though they are 
Just had this happen, I did notice once the drive is reconnected and I open the folder manually all the open files I had were re-opened.

**Comment by dogmatic69** [comment]
Another problem while using network drives is that the go to file `ctrl+p` is very slow, pretty much unusable.

**Comment by Sublime45** [comment]
This happens a lot while working on my computer (not a network hard drive).
To make things worse, it flushes any "undo history" I may have. So even if I go through the compare and accept changes, then I cannot undo any of the stuff.

It is starting to get a bit frustrating. You should at least add a setting to disable it.

**Comment by andrewdigitalchaosca** [comment]
This is happening very frequently to me as well, working on a network drive as well, every other save I get an error

**Comment by pmhoudry** [comment]
I got the same frequent error on a samba share via virtualbox host only network.

**Comment by bpasero** [comment]
I pushed a fix for the annoying save error, happy for feedback in todays insider build.

**Comment by pmhoudry** [comment]
Great ! For me however, no errors since 1.8.1

2017-01-04 8:11 GMT+01:00 Benjamin Pasero <notifications@github.com>:

**Comment by fegumi2003** [comment]
Hi all - I've been using VS Code for a week. I'm getting this error 4 out 10 times I save the file. This happens specifically with files in a SharePoint.

**Comment by bpasero** [comment]
@fegumi2003 
Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: http://code.visualstudio.com/Download#insiders

**Comment by fegumi2003** [comment]
@bpasero I've been able to reproduce this using the insider version. What troubles me is that when I'm comparing the disk vs VS Code versions; discard or accept the changes does exactly the same (changes wiped out).

**Comment by bpasero** [comment]
Ok thanks for trying, I think this needs another look then. What I fixed specifically was this error happens when you have a slow connection and manage to hit `Ctrl+S` multiple times without the first save actually returning. There might still be a bug that the `mtime` we get back after the save is not the same as the one on the server. Though still hard for me to understand how that can happen.

**Comment by rbtbar** [comment]
Regarding the issue described as "we prevent saving when we think the contents on disk are newer":

It happens to me on a LOCAL drive too, when for instance a js beautifier is triggered in the background via monitoring script.

**Comment by bpasero** [comment]
We fixed that one for 1.9.

**Comment by dogmatic69** [comment]
> We fixed that one for upcoming 1.9 release.

Just local files?

**Comment by bpasero** [comment]
Only one way to find out: http://code.visualstudio.com/Download#insiders

**Comment by meuwka** [comment]
after upgrade to 1.9 "contents on disk are newer" is still happening for me :(

**Comment by bpasero** [comment]
@meuwka when you then compare with the contents on disk, are there actually changes?

**Comment by meuwka** [comment]
@bpasero Yes, probably I found: our project use ESLint with `fix` in config (for example add spaces). Is there a possibility to disable VS Code compare mode after save file?

**Comment by bpasero** [comment]
Currently not and not sure this is the right fix. When we find the version on disk to be newer compared to the buffer in the editor we need to deal with it one way or the other to prevent dirty writes.

**Comment by meuwka** [comment]
understand, but now it's very time consuming process (many extra clicks) and doesn't work undoing changes after comparing :(

**Comment by bpasero** [comment]
Yeah, I still have to find out how it is possible that the file is modified after it has been saved from the editor.

**Comment by origin1tech** [comment]
May have to dig into this myself as it is driving me nuts. Normally use local external but recently had to migrate some projects to our NAS and now I get bad file descriptor errors like every other save. About to lose my mind!! Yes this is my worthless contribution to this thread as my post has no value whatsoever other than I feel better now for whining. :)

**Comment by bpasero** [comment]
Can you share the full error message?

**Comment by bpasero** [comment]
People here that suffer from the sporadic error when saving and we think the file is newer ( @meuwka @rbtbar @DanZeuss): I have created a new insiders build with some additional logging to find out what is going on:
* Windows: https://az764295.vo.msecnd.net/insider/5f545975b9f3c62083db0bcdbbcf54d1cc4e666a/VSCode-win32-1.11.0-insider.zip
* Linux: https://az764295.vo.msecnd.net/insider/5f545975b9f3c62083db0bcdbbcf54d1cc4e666a/code-insider-1.11.0-1488367894_amd64.tar.gz
* macOS: 
https://az764295.vo.msecnd.net/insider/5f545975b9f3c62083db0bcdbbcf54d1cc4e666a/VSCode-darwin-insider-unsigned.zip

Please start the instance from the command line passing in the `--verbose` flag. On macOS, you can directly run the executable like this: `./VS Code.app/Contents/MacOS/electron --verbose`. Once you hit the save error, please open the console (Help > Toggle Developer Tools > Console) and copy/paste the output to some location where I can access it (secret Gist is fine, the output contains the contents of the file!). Ideally you would try to reproduce with a small file to have less verbose output.

Here is a  screenshot how the error would be printed like:

![image](https://cloud.githubusercontent.com/assets/900690/23458670/8015469c-fe7d-11e6-9038-8ec72fde4073.png)

Thanks!

**Comment by pouyana** [comment]
Hi There,
I tried the the insider version on Ubuntu 16.04. Sadly for `gvfs`, the error persist:

```
Version 1.11.0-insider
Commit 0bc62d1679fbb59a4bd8f62203743408893ba4ee
Date 2017-03-02T13:49:45.971Z
Shell 1.4.6
Renderer 53.0.2785.143
Node 6.5.0
```
```
ipc.cp.ts:152 [IPC Library: Watcher] Error: watch /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/venv/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/packages/ssl_match_hostname/__init__.py ENOSPC(anonymous function) @ ipc.cp.ts:152e.invoke @ callbackList.ts:63e.fire @ event.ts:118r @ event.ts:12emitTwo @ events.js:106emit @ events.js:191process.nextTick @ internal/child_process.js:744_combinedTickCallback @ internal/process/next_tick.js:67_tickCallback @ internal/process/next_tick.js:98
ipc.cp.ts:152 [IPC Library: Watcher] Error: watch /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/venv/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/packages/ssl_match_hostname/_implementation.py ENOSPC(anonymous function) @ ipc.cp.ts:152e.invoke @ callbackList.ts:63e.fire @ event.ts:118r @ event.ts:12emitTwo @ events.js:106emit @ events.js:191process.nextTick @ internal/child_process.js:744_combinedTickCallback @ internal/process/next_tick.js:67_tickCallback @ internal/process/next_tick.js:98
ipc.cp.ts:152 [IPC Library: Watcher] Error: watch /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/pantalaimon/venv/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/packages/ssl_match_hostname/_implementation.pyc ENOSPC
extensionHost.ts:282[Extension Host] Error: This socket is closed
    at Socket._writeGeneric (net.js:673:19)
    at Socket._write (net.js:725:8)
    at doWrite (_stream_writable.js:307:12)
    at writeOrBuffer (_stream_writable.js:293:5)
    at Socket.Writable.write (_stream_writable.js:220:11)
    at Socket.write (net.js:651:40)
    at sendCommand (/home/user/.vscode-insiders/extensions/donjayamanne.python-0.5.9/out/client/providers/jediProxy.js:419:20)
    at JediProxy.sendCommand (/home/user/.vscode-insiders/extensions/donjayamanne.python-0.5.9/out/client/providers/jediProxy.js:128:16)
    at JediProxyHandler.sendCommand (/home/user/.vscode-insiders/extensions/donjayamanne.python-0.5.9/out/client/providers/jediProxy.js:548:24)
    at PythonHoverProvider.<anonymous> (/home/user/.vscode-insiders/extensions/donjayamanne.python-0.5.9/out/client/providers/hoverProvider.js:103:54)d.logExtensionHostMessage @ extensionHost.ts:282
8ipc.cp.ts:152[IPC Library: Watcher] Error: watch /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/.git/index.lock ENOSPC(anonymous function) @ ipc.cp.ts:152
ipc.cp.ts:152[IPC Library: Watcher] Error: watch /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/.git/FETCH_HEAD ENOSPC(anonymous function) @ ipc.cp.ts:152
2ipc.cp.ts:152[IPC Library: Watcher] Error: watch /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/.git/index.lock ENOSPC(anonymous function) @ ipc.cp.ts:152
ipc.cp.ts:152[IPC Library: Watcher] Error: watch /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/.git/FETCH_HEAD ENOSPC(anonymous function) @ ipc.cp.ts:152
2ipc.cp.ts:152[IPC Library: Watcher] Error: watch /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/.git/index.lock ENOSPC```

**Comment by bpasero** [comment]
@pouyana great that you can still reproduce, but I am missing the output logging I added. Are you sure you got this message showing up that the contents on disk are newer than what you have? Can you show a screenshot when this happens?

**Comment by pouyana** [comment]
@bpasero so i tried to run the vscode with --verbose, here is the output:
```
ipc.cp.ts:152[IPC Library: Watcher] Error: watch /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs//venv/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/packages/ssl_match_hostname/__init__.pyc ENOSPC(anonymous function) @ ipc.cp.ts:152
ipc.cp.ts:152[IPC Library: Watcher] Error: watch /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs//venv/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/packages/ssl_match_hostname/_implementation.pyc ENOSPC(anonymous function) @ ipc.cp.ts:152
ipc.cp.ts:152[IPC Library: Watcher] Error: watch /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs//venv/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/packages/ssl_match_hostname/__init__.py ENOSPC(anonymous function) @ ipc.cp.ts:152
ipc.cp.ts:152[IPC Library: Watcher] Error: watch /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs//.git/index.lock ENOSPC(anonymous function) @ ipc.cp.ts:152
ipc.cp.ts:152[IPC Library: Watcher] [ADDED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs//.git/index.lock
ipc.cp.ts:152[IPC Library: Watcher]  >> normalized [ADDED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs//.git/index.lock
ipc.cp.ts:152[IPC Library: Watcher] [DELETED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs//.git/index.lock
ipc.cp.ts:152[IPC Library: Watcher]  >> normalized [DELETED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/.git/index.lock
ipc.cp.ts:152[IPC Library: Watcher] [CHANGED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/package.py
ipc.cp.ts:152[IPC Library: Watcher]  >> normalized [CHANGED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/package.py
ipc.cp.ts:152[IPC Library: Watcher] [DELETED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/package.py
ipc.cp.ts:152[IPC Library: Watcher]  >> normalized [DELETED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/package.py
ipc.cp.ts:152[IPC Library: Watcher] [ADDED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/.git/index.lock
ipc.cp.ts:152[IPC Library: Watcher]  >> normalized [ADDED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/.git/index.lock
ipc.cp.ts:152[IPC Library: Watcher] [CHANGED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/.git/index.lock
ipc.cp.ts:152[IPC Library: Watcher] [CHANGED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/.git/index
ipc.cp.ts:152[IPC Library: Watcher]  >> normalized [CHANGED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/.git/index.lock
ipc.cp.ts:152[IPC Library: Watcher]  >> normalized [CHANGED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/.git/index
ipc.cp.ts:152[IPC Library: Watcher] [DELETED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/.git/index.lock
ipc.cp.ts:152[IPC Library: Watcher]  >> normalized [DELETED] /run/user/1001/gvfs/sftp:host=app.test,user=vagrant/srv/www/app/htdocs/.git/index.lock```
Hopefully it includes your chnages!

**Comment by bpasero** [comment]
@pouyana to find out you need to come to the point where the following error shows up and only then check for the console:

![image](https://cloud.githubusercontent.com/assets/900690/23516002/2f2a4274-ff6c-11e6-9521-20d5cd935de6.png)

**Comment by pouyana** [comment]
@bpasero Sadly, I don't see the error, but still the tab closes after clicking` crtl+s` more than 2 or 3 times randomly. I uploaded the whole log, if it helps.

[vscode_log.txt (console log)](https://github.com/Microsoft/vscode/files/814429/vscode_log.txt)
[terminal.txt (the log written in shell)](https://github.com/Microsoft/vscode/files/814434/terminal.txt)

**Comment by bpasero** [comment]
@pouyana but you see the error from my screenshot in https://github.com/Microsoft/vscode/issues/13665#issuecomment-283699648 ?

**Comment by pouyana** [comment]
@bpasero no. No popups and no other errors. The tab vanishes without silently. Do I have the right insider version?
```
Version 1.11.0-insider
Commit 0bc62d1679fbb59a4bd8f62203743408893ba4ee
Date 2017-03-02T13:49:45.971Z
Shell 1.4.6
Renderer 53.0.2785.143
Node 6.5.0
```

**Comment by bpasero** [comment]
@pouyana sorry I was not clear with my question, I wanted to hunt down the issue related to not being able to save. As for the issue of tabs randomly closing I actually have an idea for a fix or setting.

**Comment by bpasero** [comment]
@pouyana for your case of tabs suddenly closing, I have a custom build ready that I would like to try on you that gives some more hints:
* macOS: [Download](https://az764295.vo.msecnd.net/insider/168640ad00b636c4bac73ded8861f6a1d1a81d0a/VSCode-darwin-insider-unsigned.zip)
* Linux: [Download](https://az764295.vo.msecnd.net/insider/168640ad00b636c4bac73ded8861f6a1d1a81d0a/code-insider-1.11.0-1488535691_amd64.tar.gz)
* Windows: [Download](https://az764295.vo.msecnd.net/insider/168640ad00b636c4bac73ded8861f6a1d1a81d0a/VSCode-win32-1.11.0-insider.zip)

Please again run it with "`--verbose`" flag from the command line and when you see a tab closing out of nowhere, please send me the full output of the console. Thanks!

**Comment by pouyana** [comment]
@bpasero So I tested your build on my Ubuntu 16.04. The randomly closing tab problem is gone 👍 ! Sadly autocomplete and intellisense are not working.  But I think it is another bug probably.
```
extensionHost.ts:282 [Extension Host] ---------------------------d.logExtensionHostMessage @ extensionHost.ts:282(anonymous function) @ extensionHost.ts:137emitTwo @ events.js:106emit @ events.js:191process.nextTick @ internal/child_process.js:744_combinedTickCallback @ internal/process/next_tick.js:67_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:282 [Extension Host] sendCommand jediProxyd.logExtensionHostMessage @ extensionHost.ts:282(anonymous function) @ extensionHost.ts:137emitTwo @ events.js:106emit @ events.js:191process.nextTick @ internal/child_process.js:744_combinedTickCallback @ internal/process/next_tick.js:67_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:282 [Extension Host] Error (sendCommand) This socket has been ended by the other partyd.logExtensionHostMessage @ extensionHost.ts:282(anonymous function) @ extensionHost.ts:137emitTwo @ events.js:106emit @ events.js:191process.nextTick @ internal/child_process.js:744_combinedTickCallback @ internal/process/next_tick.js:67_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:282 [Extension Host] Error: This socket has been ended by the other party
    at Socket.writeAfterFIN [as write] (net.js:286:12)
    at sendCommand (/home/user/.vscode-insiders/extensions/donjayamanne.python-0.5.9/out/client/providers/jediProxy.js:419:20)
    at JediProxy.sendCommand (/home/user/.vscode-insiders/extensions/donjayamanne.python-0.5.9/out/client/providers/jediProxy.js:128:16)
    at JediProxyHandler.sendCommand (/home/user/.vscode-insiders/extensions/donjayamanne.python-0.5.9/out/client/providers/jediProxy.js:548:24)
    at PythonDefinitionProvider.provideDefinition (/home/user/.vscode-insiders/extensions/donjayamanne.python-0.5.9/out/client/providers/definitionProvider.js:41:38)
    at /home/user/Downloads/VSCode-linux-x64/resources/app/out/vs/workbench/node/extensionHostProcess.js:12:31393
    at /home/user/Downloads/VSCode-linux-x64/resources/app/out/vs/workbench/node/extensionHostProcess.js:6:17637
    at new n.Class.derive._oncancel (/home/user/Downloads/VSCode-linux-x64/resources/app/out/vs/workbench/node/extensionHostProcess.js:6:13740)
    at Object.f [as asWinJsPromise] (/home/user/Downloads/VSCode-linux-x64/resources/app/out/vs/workbench/node/extensionHostProcess.js:6:17600)
    at e.provideDefinition (/home/user/Downloads/VSCode-linux-x64/resources/app/out/vs/workbench/node/extensionHostProcess.js:12:31347)
```

**Comment by bpasero** [comment]
@pouyana that is good news, please keep me updated how it goes over the next couple of days and then I can turn it into a real fix.

**Comment by bpasero** [comment]
Some updates to this issue:
* before closing an editor we now do an additional check if the file is really deleted and only then close the editor (this helps for randomly closing editors after save)
* on top of that, I introduced a new setting to disable closing of deleted files entirely (see https://github.com/Microsoft/vscode/issues/11642#issuecomment-284359219 for more info)
* finally, I believe the sporadic "contents on disk are newer" errors when saving have also been resolved with my recent changes to the file model synchronisation

**Comment by Alex360hd** [comment]
Working with remote drive with VS code is actually not possible (too much bugs), I think it would be better to work on a remote explorer directly integrated with VS code.

**Comment by pthorsson** [comment]
> Another problem while using network drives is that the go to file ctrl+p is very slow, pretty much unusable.

Yeah, this is really bad. At work we're using network drives with fairly complex file structures and the **go to file** function (`ctrl + p`) takes over 5 minutes to find a file.

**Comment by DanZeuss** [comment]
To me, it was a problem just when working on VMWare on drives shared in
Linux to Windows, however, I'm using Docker with Virtual Box and I'm no
longer getting this error.

**Comment by wesruv** [comment]
A month or two ago working over my network on a Samba share was _flawless_. Using Windows 10 and a Mounted Network Drive

Now:
* `Ctrl + p` is unusable (as previously mentioned)
* Find in Folder is unusable
* Since latest update (today or yesterday) I'm getting (deleted from disk) in tabs _constantly_

**Comment by paulej** [comment]
As @wesruv mentioned, I too am seeing "(deleted from disk)" in the tabs constantly, yet clicking on the tab causes it to go back to normal.

**Comment by bpasero** [comment]
Sorry to hear that. We mark a file as deleted when we receive a file-delete event from our watcher. I will check again if there are other cases where we could report this wrongly.

**Comment by bpasero** [comment]
@wesruv can you please open a separate issue on the search issues and quick open? We recently changed our search mechanism to use RipGrep (which btw you can disable via `"search.useRipgrep": false`).

**Comment by bpasero** [comment]
I have a fix in mind to get rid of the ugly "deleted from disk" issues people are seeing, stay tuned.

**Comment by pthorsson** [comment]
@bpasero I tried `"search.useRipgrep": false`, but there wasn't any notable change i believe.

@DanZeuss I just found out that we're using VMWare too. I haven't tried any other shared drive though.

**Comment by bpasero** [comment]
Todays insider build contains a fix for the "deleted on disk" annotation showing up. Let me know how it goes: http://code.visualstudio.com/Download#insiders

**Comment by jcrben** [comment]
Seeing the content on disk is newer message too running 1.12.1, mostly when working with files that have been changed due to git. It would be helpful if the file was refreshed to show its state on the disk automatically rather than waiting until after I start edit it.

**Comment by paulej** [comment]
With the latest update, I no longer see the "deleted on disk" message.

**Comment by DanZeuss** [comment]
another problem that I'm getting that may be related to when working with
network drives is that, the file search is not working. I don't know if
it's because permission

**Comment by bpasero** [comment]
@DanZeuss please report search issues separately, thanks.

**Comment by DanZeuss** [comment]
I apologize. I thought it were some issue related only when working with
network drives.

**Comment by AdamBJ** [comment]
@bpasero I'm still getting the sporadic saving error using version 1.13.1. Is the saving problem still an ongoing issue?

**Comment by bpasero** [comment]
@AdamBJ what is your setup? and when you then compare files, is there a change to the remote file?

**Comment by AdamBJ** [comment]
I'm running Ubuntu 14.04 LTS as a virtual machine (via VMWare Workstation Player). The problem occurs when I've SSHed into a machine at my school and am working in one of the folders stored on the remote machine. As for the compare, the file on disk that is "newer" is the previous version of the file (i.e. the file minus any changes I've made since opening it). 

Let me know if you want more specific details on anything.

**Comment by se-m** [comment]
Hi.
I prefer to mount network shares to folders.
e.g. `mklink /d c:\proj \\server\share\foldername\`
I was faced with a problem that if you open such folder the editor does not read its settings from .vscode at all. But with same share folder mounted as disk vscode works fine.

Is there a way to force vscode work with such folders as with a network drive? (maybe list of folders in user settings)

VsCode:  1.13.1
Windows 7

**Comment by samirotiv** [comment]
Intellisense & autocomplete don't work on the latest version with an SMB share.

**Comment by AdamBJ** [comment]
@bpasero The network drives save issue only seems to happen when I'm using my laptop which is running a 32-bit version of VS Code on Ubuntu 14.04 LTS.

**Comment by RobIsHere** [comment]
https://github.com/Microsoft/vscode/issues/31667

**Comment by nguyenbathanh** [comment]
Any news?

**Comment by barberdt** [comment]
I can confirm @samirotiv's statement regarding autocomplete/intellisense not working with an SSHFS-mounted drive (on Mac). Similar to @nguyenbathanh, I'm curious what may be going on here. FWIW, the [Path Intellisense](https://marketplace.visualstudio.com/items?itemName=christian-kohler.path-intellisense) extension seems to have no issue auto-completing paths with the same mounted drive. Thanks for any and all support!

**Comment by barberdt** [comment]
@bpasero is this still the proper place to bring up related issues or would a new issue make more sense? Happy to accommodate in whatever way makes the most sense. Thanks!

**Comment by ChadRidings** [comment]
Error: Failed to save 'syncSummary.txt': The content on disk is newer. Click on Compare to compare your version with the one on disk. (Settings Sync 2.8.2 on Dec 18, 2017)

**Comment by paulej** [comment]
@ChadRidings Is that a Visual Studio Code issue or a Settings Sync issue (https://marketplace.visualstudio.com/items?itemName=Shan.code-settings-sync).  That repo is https://github.com/shanalikhan/code-settings-sync/

**Comment by ChadRidings** [comment]
@paulej that is a good question. Yea I was looking at that repo yesterday as well. I've been taken down a rabbit hole where it's hard for me to determine if this is a problem with the application or a problem with the extension. I'll report back when I have a better idea.

**Comment by THCloud** [comment]
hi：I also have similar problem， I work with samba， open a folder on remote machine， （mine is Win， remote‘s is linux）， and I wanna use shell on remote machine， can VSCode integrated teminal do it？

**Comment by DanZeuss** [comment]
@THCloud yes, you can, but I strongly recommend you to use the integrated terminal by SSH, I mean, connect to the server using SSH session.

**Comment by dotnetjalps** [comment]
Is there any update for this? As I m using it for Linux network drive and it's taking so much time compared to sublime text. Sublime text it's blazing fast.

**Comment by aimcom** [comment]
I have problems with sshfs mounted directories in Visual Studio Code on Ubuntu, too. Autocomplete and IntelliSense don't work properly.

So I've created a small **workaround** that I want to share: It worked for me to copy the sshfs directory locally, watch changes in this directory with **[inotifywait](https://linux.die.net/man/1/inotifywait)** and copy them into the sshfs directory with a small shell script. This doesn't reflect changes on the remote filesystem but this is not important for me.

**Comment by johndugan** [comment]
Another Samba user here. I have two workflows - If I'm remote, I work with a Debian VM on Hyper-V (on my laptop). If I am at my desk, I work with a Debian VM on a VMware server that has better hardware.

In both cases, the Git features in code have very poor performance. Search is fairly performant though.

**Comment by gbrent** [comment]
I was trying to use VS Code with my git repository on a samba share. I am using mac and the samba share is on a remote Ubuntu 18.04 machine hosted on AWS. VS code and git would not work properly when I had my repository on a samba share but I had to get work done so I went ahead and started working without git. I was editing a file that was 2 days old, hours of work, and VS code hanged. When I reopened VS Code I was shocked to see my file on the server was blank. In fact, every file I had open at the time of the crash was blank. Now if git would have been workign properly I would not have lost data. I guess it is back to Dreamweaver?
